packages/back-end/src/services/features.ts

```diff
@@ -1047,3 +1151,255 @@ export const updateInterfaceEnvSettingsFromApiEnvSettings = (
     };
   }, existing);
 };
+
+// Only keep features that are "on" or "conditional". For "on" features, remove any top level prerequisites
+export const reduceFeaturesWithPrerequisites = (
+  features: FeatureInterface[],
+  environment: string,
+  prereqStateCache: Record<string, Record<string, PrerequisiteStateResult>> = {}
+): FeatureInterface[] => {
+  if (!prereqStateCache[environment]) {
+    prereqStateCache[environment] = {};
+  }
+
+  const newFeatures: FeatureInterface[] = [];
+
+  // block "always off" features, or remove "always on" prereqs
+  for (const feature of features) {
+    const newFeature = cloneDeep(feature);
+    let removeFeature = false;
+
+    const newPrerequisites: FeaturePrerequisite[] = [];
+    for (const prereq of newFeature.prerequisites || []) {
+      let state: PrerequisiteStateResult = {
+        state: "deterministic",
+        value: null,
+      };
+      if (prereqStateCache[environment][prereq.id]) {
+        state = prereqStateCache[environment][prereq.id];
+      } else {
+        const prereqFeature = features.find((f) => f.id === prereq.id);
+        if (prereqFeature) {
+          state = evaluatePrerequisiteState(
+            prereqFeature,
+            features,
+            environment,
+            undefined,
+            true
+          );
+        }
+        prereqStateCache[environment][prereq.id] = state;
+      }
+
+      switch (state.state) {
+        case "conditional":
+          // keep the feature and the prerequisite
+          newPrerequisites.push(prereq);
+          break;
+        case "cyclic":
+          removeFeature = true;
+          break;
+        case "deterministic":
+          // todo: this makes the assumption that top-level prereqs are always boolean / true checks
+          if (state.value === null || state.value === "false") {
```

Comment:
This assumption does not seem safe.  I think we need the same logic as rules where we look at the specific condition and determine if it always passes, always fails, or is conditional.

Reply:
replaced with evalCondition()